analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/AnalysisScaffolding.cs

```diff
@@ -80,6 +81,42 @@ public static string CreateSonarProjectConfigWithFilesToAnalyze(TestContext cont
         public static string CreateSonarProjectConfig(TestContext context, ProjectType projectType, bool isScannerRun = true) =>
             CreateSonarProjectConfig(context, "ProjectType", projectType.ToString(), isScannerRun);
 
+        public static string CreateSonarLintXml(
+            TestContext context,
+            string language = LanguageNames.CSharp,
+            bool analyzeGeneratedCode = false,
+            string[] exclusions = null,
+            string[] inclusions = null,
+            string[] globalExclusions = null,
+            string[] testExclusions = null,
+            string[] testInclusions = null,
+            string[] globalTestExclusions = null) =>
+            TestHelper.WriteFile(context, "SonarLint.xml", GenerateSonarLintXmlContent(language, analyzeGeneratedCode, exclusions, inclusions, globalExclusions, testExclusions, testInclusions, globalTestExclusions));
+
+        public static string GenerateSonarLintXmlContent(
+            string language = LanguageNames.CSharp,
+            bool analyzeGeneratedCode = false,
+            string[] exclusions = null,
+            string[] inclusions = null,
+            string[] globalExclusions = null,
+            string[] testExclusions = null,
+            string[] testInclusions = null,
+            string[] globalTestExclusions = null) =>
+            new XDocument(
+                new XDeclaration("1.0", "utf-8", "yes"),
+                new XElement("AnalysisInput",
+                    new XElement("Settings",
+                        CreateKeyValuePair("Setting", $"sonar.{(language == LanguageNames.CSharp ? "cs" : "vbnet")}.analyzeGeneratedCode", analyzeGeneratedCode.ToString()),
```

Comment:
Also if you create a method called CreateSetting, then you don't have to repeat the "Setting" string 7 times.